chainservice/builder.go

```diff
@@ -582,6 +582,9 @@ func (builder *Builder) registerStakingProtocol() error {
 
 func (builder *Builder) registerRewardingProtocol() error {
 	// TODO: rewarding protocol for standalone mode is weird, rDPoSProtocol could be passed via context
+	if builder.cfg.Consensus.Scheme != config.RollDPoSScheme {
+		return nil
```

Comment:
should panic if being used falsely?

Reply:
in non-rolldpos mode, we can simply not to register the rewarding protocol. do you mean to limit from the caller side?

Reply:
doing it outside looks more readable for me
```
if builder.cfg.Consensus.Scheme == config.RollDPoSScheme {
builder.registerRewardingProtocol()
}
```